File: demos/jans-chip/ios/Jans-Chip/Database/DBModel/AppIntegrityEntity.swift
//
//  AppIntegrityEntity.swift
//  Jans-Chip
//
//  Created by Nazar Yavornytskyi on 12.10.2023.
//

import RealmSwift

final class AppIntegrityEntity: Object, ObjectKeyIdentifiable {
    
    override static func primaryKey() -> String? {
        "id"
    }
    
    @Persisted var id: String = "1"
    @Persisted var sno: String
    @Persisted var appIntegrity: String
    @Persisted var deviceIntegrity: String
    @Persisted var appLicensingVerdict: String
    @Persisted var requestPackageName: String
    @Persisted var nonce: String
    @Persisted var error: String
}

File: demos/jans-chip/kmp/fido2/src/commonMain/kotlin/com/example/fido2/model/appIntegrity/AppIntegrityEntity.kt
package com.example.fido2.model.appIntegrity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.SerialName

@Entity(tableName = "APP_INTEGRITY")
data class AppIntegrityEntity(

    @PrimaryKey
    @SerialName("SNO")
    var sno: String,

    @ColumnInfo(name = "APP_INTEGRITY")
    var appIntegrity: String? = null,

    @ColumnInfo(name = "DEVICE_INTEGRITY")
    var deviceIntegrity: String? = null,

    @ColumnInfo(name = "APP_LICENSING_VERDICT")
    var appLicensingVerdict: String? = null,

    @ColumnInfo(name = "REQUEST_PACKAGE_NAME")
    var requestPackageName: String? = null,

    @ColumnInfo(name = "NONCE")
    var nonce: String? = null,

    @ColumnInfo(name = "ERROR")
    var error: String? = null,
)